WordPress/Sniffs/VIP/SuppressFiltersSniff.php

```diff
@@ -0,0 +1,192 @@
+
+						$keyIdx = $this->phpcsFile->findPrevious( array( T_WHITESPACE, T_CLOSE_SQUARE_BRACKET ), ( $operator - 1 ),null,true );
+
+						if ( ! is_numeric( $this->tokens[ $keyIdx  ]['content'] ) ) {
+							$key            = $this->strip_quotes( $this->tokens[ $keyIdx ]['content'] );
+							$valStart       = $this->phpcsFile->findNext( array( T_WHITESPACE ), ( $operator + 1 ), null, true );
+							$valEnd         = $this->phpcsFile->findNext( array( T_COMMA, T_SEMICOLON ), ( $valStart + 1 ), null, false, null, true );
+							$val            = $this->phpcsFile->getTokensAsString( $valStart, ( $valEnd - $valStart ) );
+							$val            = $this->strip_quotes( $val );
+							$array_value = $val;
+						}
+					} elseif ( in_array( $this->tokens[ $varData ]['code'], array( T_ARRAY, T_OPEN_SHORT_ARRAY ), true ) ) {
+						// Store array data into variable so that we can proceed later.
+						$param_items = $this->get_function_call_parameters( $varData );
+					}
+				}
+			}
+		} elseif ( T_ARRAY === $this->tokens[ $argument_data ]['code'] || T_OPEN_SHORT_ARRAY === $this->tokens[ $argument_data ]['code'] ) {
+
+			// It covers multiple array variable directly passed to function.
+			// eg. get_posts( array( 'foo' => 'bar', 'baz' => 'quux' ) ).
+			$param_items = $this->get_function_call_parameters( $argument_data );
+
+		} elseif ( T_CONSTANT_ENCAPSED_STRING === $this->tokens[ $argument_data ]['code'] || T_DOUBLE_QUOTED_STRING === $this->tokens[ $argument_data ]['code'] ) {
+
+			// It handles when key value comes in '&query=arg' format.
+			// eg. get_posts( 'foo=bar&baz=quux' ).
+			if ( preg_match_all( '#(?:^|&)([a-z_]+)=([^&]*)#i', $this->strip_quotes( $this->tokens[ $argument_data ]['content'] ), $arrKey ) <= 0 ) {
+				return;
+			}
+
+			foreach ( $arrKey[1] as $i => $query_key ) {
+
+				if ( 'suppress_filters' === $query_key ) {
+
+					$is_used     = true;
+					$array_value = $arrKey[2][ $i ];
+				}
+			}
+		}
+
+		// Process multi dimensional array.
+		if ( ! empty( $param_items ) ) {
+
+			// Process the array.
+			foreach ( $param_items as $item ) {
+
+				// If 'raw' contains suppress_filters value.
+				if ( false !== strpos( $item['raw'], 'suppress_filters' ) ) {
+
+					// Finding the value by token pointer.
+					for ( $ptr = $item['start']; $ptr <= $item['end']; $ptr++ ) {
+
+						// '=>' detected.
+						if ( T_DOUBLE_ARROW === $this->tokens[ $ptr ]['code'] ) {
+
+							$accept_type = array( T_DOUBLE_QUOTED_STRING, T_CONSTANT_ENCAPSED_STRING, T_FALSE, T_TRUE, T_LNUMBER, T_INT_CAST );
+
+							$key_value = $this->phpcsFile->findNext( Tokens::$emptyTokens, $ptr + 1, null, true );
+
+							if ( in_array( $this->tokens[ $key_value ]['code'], $accept_type,true ) ) {
+
+								$is_used = true;
+								$array_value = $this->strip_quotes( $this->tokens[ $key_value ]['content'] );
+							}
+						}
+					}
+				}
+			}
+		}
+
+		// If so, expected value was not passed.
+		if ( ! $is_used || ! empty( $array_value ) ) {
```

Comment:
Being explicit is helpful here: `false === $is_used`.

[Edit]: on second thought: this condition will always be `true` as either `$is_used` will be `false` or `$array_value` will be filled, so should be removed or am I missing something ?